meson.build

```
@@ -355,6 +351,16 @@ if enable_introspection
   dependency('gobject-introspection-1.0', version: '>= 0.6.2')
 endif
 
+config_h = configure_file(
+  output: 'config.h',
+  configuration: config_data,
+)
+
+compiler_common_flags += ['-include', config_h.full_path()]
+
+add_project_arguments(compiler_common_flags + compiler_c_flags, language: 'c')
+add_project_arguments(compiler_common_flags + compiler_cpp_flags, language: 'cpp')
```

I think this can be simplified a bit:
 - there's no C++ code, so the second `add_project_arguments()` invocation can be dropped
 - no need to keep the CFLAGS in a separate array, just use `add_project_arguments()` directly, since it appends the argument(s)
 - you don't need to use the full path to the `config.h` file, so the related meson version bump can be dropped

```suggestion
add_project_arguments(['-include', 'config.h'], language: 'c')
```

---

Apologies, I completely missed the fact that the `compiler_c_flags` stuff is preexisting (and that there's at least one cpp file). In that case ignore the previous two points.

As for the last one, I think the meson bump is too big for a very little gain, but I'll leave the decision to @jrybar-rh.

---

Hello, IMO there is a possibility of regrets when we sacrifice CI for F39. I know, just 5 months, but if sith happens, testing fixes for F39 would be unnecessarily difficult. Can we make the 'bump-needing' feature postponed (and reminded by a new issue)?

---

Okay, I'll make a new PR for it once this is merged.

---

See #465